src/Engine/MapInfo.cpp

```diff
@@ -251,19 +251,19 @@ void MapInfo::SpawnRandomTreasure(SpawnPoint *a2) {
     ITEM_TREASURE_LEVEL v13 = Sample(RemapTreasureLevel(a2->uItemIndex, Treasure_prob));
     if (v13 != ITEM_TREASURE_LEVEL_GUARANTEED_ARTIFACT) {
         // [0, 20) -- nothing
-        // [20, 60) -- only gold
-        // [60, 100) -- gold & item
-        // TODO(captainurist): code in GenerateItemsInChest doesn't generate gold on [60, 100)
+        // [20, 60) -- gold
+        // [60, 100) -- item
 
         if (v5 < 20)
             return;
 
         if (v5 >= 60) {
-            DropTreasureAt(v13, Random(27) + 20,
-                                               a2->vPosition.x,
+            DropTreasureAt(v13, Random(27) + 20, a2->vPosition.x,
                                                a2->vPosition.y,
                                                a2->vPosition.z, 0);
+            return;
```

Comment:
We're changing behavior here. Is the new behavior the same as it was in vanilla?

Reply:
Returning behavior to vanilla. Shouldn't drop an item and gold.

Reply:
Got it!